README.md

```diff
@@ -63,6 +63,30 @@ cep('05010000')
   // }
 ```
 
+### Realizando uma consulta avançada
+Para passar configurações como Proxy e Agent:
+
+
+``` js
+import cep from 'cep-promise'
+import https from 'https'
+
+const agent = new https.Agent({ keepAlive: true })
+const proxyURL = 'socks5://127.0.0.1:1950'
+
+const configurations = { agent, proxyURL }
```

Comment:
@LorhanSohaky 

Aproveitaria este PR e adicionaria novas configurações, como o timeout da requisição, que já foi adicionada pelo @filipedeschamps no PR #190 e também headers, como o User-Agent (para quando rodar do lado do servidor) que eu ajustei no PR #189. 

Não esquecer do valor default destas novas configurações.

Dessa forma podemos cancelar os PRs anteriores e resolver tudo neste!

Reply:
@ivmarcos , para fazer isso, precisamos decidir se serão passadas as mesmas configurações para todos os serviços ou cada um pode receber uma configuração diferente (vide exemplo #190). E se for um para cada provedor, qual será o formato para isso.

O que tenho em mente é fazer algo parecido com:
```js
const config = {
  providersConfig: {
    brasilapi: {
        timeout: 1000,
        proxyURL: '...'
    }
  }
}
```

Reply:
@LorhanSohaky

Exatamente! Cada provider precisa receber configurações específicas, mas acredito que poderia ter uma opção pra definir geral:

`const config = {
		providersConfig: {
			//specific providers
			brasilapi: {
				timeout: 1000,
				proxyURL: '...',
				headers: {
					'user-agent': 'especific ua'
				}
			},
			// for all providers
			headers: {
				'user-agent': 'cep-promise'
			}
		}
	}`

Reply:
Modifiquei, agora suporta sobrescrever os headers e adicionar timeout

Reply:
@ivancorrea consegue fazer o review?